R/representativity.R

```diff
@@ -0,0 +1,102 @@
+
+  indx <- lt.indices(life_table)
+
+  # Criteria for representativity after Weiss 1973
+
+  representativity_sum <- 0
+
+  weiss_i1 <-  if (indx$mortality$q10_5 < indx$mortality$q15_5) {
+    representativity_sum <- representativity_sum + 0
+    crayon::green(paste0("5q10 = ", round(indx$mortality$q10_5, 1), " < 5q15 = ",
+                         round(indx$mortality$q15_5, 1),"."))
+  } else {
+    representativity_sum <- representativity_sum + 1
+    crayon::red(paste0("5q10 = ", round(indx$mortality$q10_5, 1), " > 5q15 = ",
+                       round(indx$mortality$q15_5, 1),"."))
+  }
+
+  weiss_i2 <- if (indx$mortality$q0_5 > indx$mortality$q15_5) {
+    representativity_sum <- representativity_sum + 0
+    crayon::green(paste0("5q0 = ", round(indx$mortality$q0_5, 1), " > 5q15 = ",
+                         round(indx$mortality$q15_5, 1),"."))
+  } else {
+    representativity_sum <- representativity_sum + 1
+    crayon::red(paste0("5q0 = ", round(indx$mortality$q0_5, 1),
+                       " < 5q15 = ", round(indx$mortality$q15_5, 1),"."))
+  }
+
+
+  # Criteria for representativity after Bocquet-Appel and Masset
+
+  child_i <- if (indx$child_i$d5_9_d10_14 < 2) {
+    representativity_sum <- representativity_sum + 1
+    crayon::red(paste0("Index of children is ", round(indx$child_i$d5_9_d10_14, 2),
+                       " and therefore under the expected value of 2."))
+  } else {
+    representativity_sum <- representativity_sum + 0
+    crayon::green(paste0("Index of children is ", round(indx$child_i$d5_9_d10_14, 2),
+                         " and therefore above the expected value of 2."))
+  }
+
+  juvenile_i <- if (indx$juvenile_i$d5_14_d20plus < 0.1) {
+    representativity_sum <- representativity_sum + 1
+    crayon::red(paste0("Index of juveniles is ", round(indx$juvenile_i$d5_14_d20plus,2),
+                       " and therefore under the expected value of 0.1."))
+  } else {
+    representativity_sum <- representativity_sum + 0
+    crayon::green(paste0("Index of juveniles is ", round(indx$juvenile_i$d5_14_d20plus,2),
+                         " and therefore above the expected value of 0.1."))
+  }
+
+  representativity_verdict <- if (representativity_sum > 1) {
+    crayon::red(paste0(representativity_sum, " of 4 criteria for representativity of subadults are NOT given. You shoud refrain from any corrections."))
+  } else if (representativity_sum == 1) {
+    crayon::red(paste0("One of 4 criteria for representativity of subadults is NOT given. You shoud be careful with any corrections."))
+  } else {
+    crayon::green(paste0("General representativity of subadults seems to be given. Further corrections might be applied."))
+  }
+
+  cat("Criteria for representativity after Weiss 1973:\n", weiss_i1, "\n", weiss_i2,
+      "\n\nCriteria for representativity after Bocquet-Appel and Masset:\n", child_i, "\n", juvenile_i,
+      "\n\nRepresentativity verdict:\n",representativity_verdict)
```

Comment:
As already mentioned above: That's not a good way to provide the output of a function.

Reply:
If you want to have this kind of message text, then you should define another list class and write a dedicated print function for it.  Then you have nice console messages AND usable numerical output.

Reply:
Sounds sensible. BUT: How would such a list class and a dedicated print function look like?

Reply:
If I understand that correctly, this function calculates a lot of different individual values. The output for example could be a list of (the new) class  `representativity_output`. 

```
foo <- list(
  index_of_children = round(indx$child_i$d5_9_d10_14, 2),
  index_of_juveniles = round(indx$juvenile_i$d5_14_d20plus,2)
) %>% `class<-`("representativity_output")

format.representativity_output <- function(x) {
  return_value <- paste(x$index_of_children, "is a rainbow in the dark.")
  invisible(return_value)
}

print.representativity_output <- function(x, ...) cat(format(x, ...), "\n")

foo
```

Take a look at R/class_functions.R for an example right in this package. More info [here](http://adv-r.had.co.nz/S3.html).

Reply:
`representativity_output` is not a good class name btw.. I guess you can come up with something better.

Reply:
Thanks for the hint! I will try it this way.

Reply:
After rethinking the matter, I opted against the dedicated class-option and sticked to a simple, but hopefully informative and clear data.frame. What do you think?